biz.aQute.bndlib/src/aQute/bnd/make/component/ServiceComponent.java

```diff
@@ -119,9 +121,73 @@ private void componentEntry(Map<String, Map<String, String>> serviceComponents,
 			boolean fqn = Verifier.isFQN(name);
 			if (fqn)
 				createComponentResource(serviceComponents, name, info);
-			else
-				error(Constants.SERVICE_COMPONENT
-					+ " entry '%s' is not a FQN. Bnd no longer supports the deprecated Bnd DS annotations.", name);
+			else {
+				reportInvalidUseOfServiceComponentHeader(name);
+			}
+		}
+
+		/*
+		 * Report the use of old bnd annotations
+		 * @param name the name used in the Service-Component headers
+		 */
+
+		private void reportInvalidUseOfServiceComponentHeader(String name) throws Exception, IOException {
+			SetLocation warning = warning(Constants.SERVICE_COMPONENT
+				+ " is normally generated by bnd. If you want to point at specific component XML files (wildcards are allowed, see findEntries) make sure it ends in '.xml'. Found %s",
+				name);
+
+			FileLine where = getHeader(Pattern.compile(Constants.SERVICE_COMPONENT),
+				Pattern.compile(Pattern.quote(name)));
+			if (where != null) {
+				where.set(warning);
+			}
+
+			warning.reference(
+				"https://osgi.org/specification/osgi.cmpn/7.0.0/service.component.html#d0e36857");
+
+			TypeRef componentClassRef = analyzer.getTypeRefFromFQN("aQute.bnd.annotation.component.Component");
+			Clazz componentClass = analyzer.findClass(componentClassRef);
+			if (componentClass == null) {
```

Comment:
But couldn't the bundle being built include a package from a jar on the classpath? This does not require the Bnd DS annotations to be on the classpath. They are only needed on the classpath to compile source not to include a package already compiled.

Reply:
A bit of an edge case I guess you're right. I can fix that later.